### DISCUSSION TITLE
Will upgrade from 2.3.0 to 3.9.3 have breaking changes?

### DISCUSSION THREAD

**Discussion by dotswing** [sidorares/node-mysql2#2540] [General]
My project is pretty old and still use version 2.3.0. Will upgrading to 3.9.3 have breaking change to some functionality?
As I look through the release notes there seem to be no breaking changes.

**Comment by sidorares** [comment]
As far as I know - no intentional backwards-incompatible changes. We had a very long pause between releases a while ago and decided to make a major release just to mark that. Note that quite a lot of dependencies use modern JS now, while 2.3.0 could be used with vey old node versions 3.x series require at least v14 ( in addition to dependencies also changes in the mysql2 codebase itself - removed `safe-buffer` shim, use of native Promise, arrow functions, classes etc )

The surface API is motly stable. Authentication plugins interface is being redesigned and old is considered deprecated ( but still there and works ), most other api changes are additions

**Reply by dotswing** [reply]
Thank for the answer @sidorares. Just wanted to make sure no breaking changes. Will test the system out. 
Thank you very much for great library.

**Comment by JerrysShan** [comment]
but version 3.x  dependencies  [lru-cache](https://github.com/isaacs/node-lru-cache/blob/v8.0.0/package.json#L81) require at least node.js >=16.14

**Reply by sidorares** [reply]
my bad, you are correct. Because of lru-cache the minimal node version for 3.x line is v16 now ( which is still very old ) while 2.x can be used with pretty much anything with `Buffer` in it, could probably run on node 0.10

**Reply by wellwelwel** [reply]
It is interesting to consider that the entire current test suite passes without errors in **Node.js 14** 🙋🏻‍♂️
